File: Python/Check_Out_App.py
from datetime import datetime

class Item:
    def __init__(self, name, quantity, price):
        self.name = name
        self.quantity = quantity
        self.price = price

    def get_total_of_purchase(self):
        return self.quantity * self.price


def get_cashier_name():
    return input("What is cashier's name? ")


def get_customer_name():
    return input("What is customer's name? ")


def collect_items():
    items = []
    while True:
        name = input("What did our customer buy? ")
        qty = int(input(f"How many pieces of {name} did our customer buy? "))
        price = float(input(f"How much is the {name} per unit? "))

        items.append(Item(name, qty, price))

        add_more = input("Add more items? (yes/no): ").strip().lower()
        if add_more != "yes":
            break
    return items


def get_discount():
    return float(input("Enter discount (%): "))


def calculate_sub_total(items):
    return sum(item.get_total_of_purchase() for item in items)


def print_header(cashier_name, customer_name):
    print("SEMICOLON STORES")
    print("MAIN BRANCH")
    print("LOCATION: 312, HERBERT MACAULAY WAY, SABO YABA, LAGOS")
    print("TEL: 03293828343")
    print("Date:", datetime.now().strftime("%d-%b-%y %I:%M:%S %p"))
    print("Cashier:", cashier_name)
    print("Customer Name:", customer_name)
    print("-------------------------------------------")
    print(f"{'ITEM':<10} {'QTY':<5} {'PRICE':<10} {'TOTAL(NGN)':<10}")


def print_items(items):
    for item in items:
        print(f"{item.name:<10} {item.quantity:<5} {item.price:<10.2f} {item.get_total_of_purchase():<10.2f}")


def print_invoice(items, sub_total, discount, vat, bill_total, cashier_name, customer_name):
    print("\n================= INVOICE =================")
    print_header(cashier_name, customer_name)
    print_items(items)
    print("-------------------------------------------")
    print(f"Sub Total: {sub_total:.2f}")
    print(f"Discount: {discount:.2f}")
    print(f"VAT @ 7.50%: {vat:.2f}")
    print("-------------------------------------------")
    print(f"Bill Total: {bill_total:.2f}")
    print("THIS IS NOT A RECEIPT. KINDLY PAY", bill_total)
    print("===========================================\n")


def get_payment():
    return float(input("How much did the customer give to you? "))


def print_receipt(items, sub_total, discount, vat, bill_total, amount_paid, balance, cashier_name, customer_name):
    print("\n================= RECEIPT =================")
    print_header(cashier_name, customer_name)
    print_items(items)
    print("-------------------------------------------")
    print(f"Sub Total: {sub_total:.2f}")
    print(f"Discount: {discount:.2f}")
    print(f"VAT @ 7.50%: {vat:.2f}")
    print("-------------------------------------------")
    print(f"Bill Total: {bill_total:.2f}")
    print(f"Amount Paid: {amount_paid:.2f}")
    print(f"Balance: {balance:.2f}")
    print("===========================================")
    print("THANK YOU FOR YOUR PATRONAGE!")


def main():
    cashier_name = get_cashier_name()
    customer_name = get_customer_name()

    items = collect_items()

    discount_percent = get_discount()
    sub_total = calculate_sub_total(items)
    discount = (discount_percent / 100) * sub_total
    vat = 0.075 * sub_total
    bill_total = (sub_total + vat) - discount

    print_invoice(items, sub_total, discount, vat, bill_total, cashier_name, customer_name)

    amount_paid = get_payment()
    balance = amount_paid - bill_total

    print_receipt(items, sub_total, discount, vat, bill_total, amount_paid, balance, cashier_name, customer_name)


if __name__ == "__main__":
    main()

File: JavaScript/CheckOutApp.js
const prompt = require("prompt-sync")();

class Item {
  constructor(name, quantity, price) {
    this.name = name;
    this.quantity = quantity;
    this.price = price;
  }
  getTotalOfPurchase() {
    return this.quantity * this.price;
  }
}

function getCashierName() {
  return prompt("What is cashier's name? ");
}

function getCustomerName() {
  return prompt("What is customer's name? ");
}

function collectItems() {
  let items = [];
  while (true) {
    let name = prompt("What did our customer buy? ");
    let qty = parseInt(prompt(`How many pieces of ${name} did our customer buy? `));
    let price = parseFloat(prompt(`How much is the ${name} per unit? `));
    items.push(new Item(name, qty, price));
    let addMore = prompt("Add more items? (yes/no): ");
    if (addMore.toLowerCase() !== "yes") break;
  }
  return items;
}

function getDiscount() {
  return parseFloat(prompt("Enter discount (%): "));
}

function calculateSubTotal(items) {
  return items.reduce((sum, item) => sum + item.getTotalOfPurchase(), 0);
}

function printInvoice(items, subTotal, discount, vat, billTotal, cashierName, customerName) {
  console.log("\n================= INVOICE =================");
  printHeader(cashierName, customerName);
  printItems(items);
  console.log("-------------------------------------------");
  console.log(`Sub Total: ${subTotal.toFixed(2)}`);
  console.log(`Discount: ${discount.toFixed(2)}`);
  console.log(`VAT @ 7.50%: ${vat.toFixed(2)}`);
  console.log("-------------------------------------------");
  console.log(`Bill Total: ${billTotal.toFixed(2)}`);
  console.log("THIS IS NOT A RECEIPT. KINDLY PAY " + billTotal.toFixed(2));
  console.log("===========================================\n");
}

function getPayment() {
  return parseFloat(prompt("How much did the customer give to you? "));
}

function printReceipt(items, subTotal, discount, vat, billTotal, amountPaid, balance, cashierName, customerName) {
  console.log("\n================= RECEIPT =================");
  printHeader(cashierName, customerName);
  printItems(items);
  console.log("-------------------------------------------");
  console.log(`Sub Total: ${subTotal.toFixed(2)}`);
  console.log(`Discount: ${discount.toFixed(2)}`);
  console.log(`VAT @ 7.50%: ${vat.toFixed(2)}`);
  console.log("-------------------------------------------");
  console.log(`Bill Total: ${billTotal.toFixed(2)}`);
  console.log(`Amount Paid: ${amountPaid.toFixed(2)}`);
  console.log(`Balance: ${balance.toFixed(2)}`);
  console.log("===========================================");
  console.log("THANK YOU FOR YOUR PATRONAGE!");
}

function printHeader(cashierName, customerName) {
  const date = new Date().toLocaleString("en-GB", { hour12: true });
  console.log("SEMICOLON STORES");
  console.log("MAIN BRANCH");
  console.log("LOCATION: 312, HERBERT MACAULAY WAY, SABO YABA, LAGOS");
  console.log("TEL: 03293828343");
  console.log("Date: " + date);
  console.log("Cashier: " + cashierName);
  console.log("Customer Name: " + customerName);
  console.log("-------------------------------------------");
  console.log("ITEM       QTY   PRICE      TOTAL(NGN)");
}

function printItems(items) {
  items.forEach(item => {
    console.log(
      `${item.name.padEnd(10)} ${item.quantity
        .toString()
        .padEnd(5)} ${item.price.toFixed(2).padEnd(10)} ${item.getTotalOfPurchase().toFixed(2)}`
    );
  });
}

function main() {
  let cashierName = getCashierName();
  let customerName = getCustomerName();
  let items = collectItems();
  let discountPercent = getDiscount();
  let subTotal = calculateSubTotal(items);
  let discount = (discountPercent / 100) * subTotal;
  let vat = 0.075 * subTotal;
  let billTotal = (subTotal + vat) - discount;
  printInvoice(items, subTotal, discount, vat, billTotal, cashierName, customerName);
  let amountPaid = getPayment();
  let balance = amountPaid - billTotal;
  printReceipt(items, subTotal, discount, vat, billTotal, amountPaid, balance, cashierName, customerName);
}

main();
